airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -296,14 +295,21 @@ private Job getJob(DSLContext ctx, long jobId) {
   }
 
   @Override
-  public List<Job> listJobs(ConfigType configType, String configId) throws IOException {
+  public List<Job> listJobs(ConfigType configType, String configId, int pagesize, int offset) throws IOException {
+    Set<ConfigType> configTypes = new HashSet<ConfigType>();
+    configTypes.add(configType);
+    return listJobs(configTypes, configId, pagesize, offset);
+  }
+
+  @Override
+  public List<Job> listJobs(Set<ConfigType> configTypes, String configId, int pagesize, int offset) throws IOException {
     return database.query(ctx -> getJobsFromResult(ctx.fetch(
         BASE_JOB_SELECT_AND_JOIN + "WHERE " +
-            "CAST(config_type AS VARCHAR) = ? AND " +
-            "scope = ? " +
-            "ORDER BY jobs.created_at DESC",
-        Sqls.toSqlName(configType),
-        configId)));
+            "CAST(config_type AS VARCHAR) in " + Sqls.toSqlInFragment(configTypes) + " " +
+            "AND scope = ? " +
+            "ORDER BY jobs.created_at DESC, jobs.id DESC, attempts.created_at ASC, attempts.id ASC " +
```

Comment:
may be worth assigning this as a constant? having to think through the ordering is not that intuitive (as evidence by the fact that the original code was wrong), so reducing the chance that it gets changed in one place and not other might be good. up to you. just a thought.